client/tasks/task-list.tsx

```diff
@@ -49,16 +49,9 @@ export const TaskList: React.FC< TaskListProps > = ( {
 	} );
 	const eventPrefix = getEventPrefix( id );
 	const prevQueryRef = useRef( query );
-	const nowTimestamp = Date.now();
-	const visibleTasks = tasks.filter(
```

Comment:
Looks much cleaner client-side!  I'm still a bit concerned that this doesn't address the original issue:

>Note that certain items like "dismiss" and "snooze" still need to optimistically update and potentially undo that update, which makes me unsure whether or not we can filter visibility on the REST side. cc @joelclimbsthings

Dismissing or snoozing tasks won't work as expected.

Filtering them out entirely may be problematic too if we undo snoozing or undo dismissal.

Reply:
Thanks @joshuatf , this is true and I've refactored the changes to leave the logic around snooze and dismissal on the client-side. I think ultimately that's the simpler and more flexible approach. The API is, however, filtering out tasks that are `canView: false` since those are never visible on the front-end, so it seems safe to take them out.